packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -0,0 +1,131 @@
+import {validateOptions} from '../../../../utils/validate-payload';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+import {loadReducerError} from '../../../../utils/errors';
+import {cartReducer as cart} from '../../../../features/commerce/context/cart/cart-slice';
+import {
+  addItem,
+  removeItem,
+  setItems,
+  updateItemQuantity,
+} from '../../../../features/commerce/context/cart/cart-actions';
+import {cartSchema} from '../../../../features/commerce/context/cart/cart-validation';
+
+export interface CartOptions {
+  items?: CartItem[];
+}
+
+export interface CartItem {
+  productId: string;
+  quantity: number;
+}
+
+export interface CartProps {
+  /**
+   * The initial options that should be applied to this `Cart` controller.
+   */
+  options?: CartOptions;
+}
+
+/**
+ * The `Cart` controller exposes methods for managing the cart in a commerce interface.
+ */
+export interface Cart extends Controller {
+  /**
+   * Sets the cart items.
+   * @param cart - The new cart.
```

Comment:
Based on this parameter description/name, I'd expect `cart` to be a Cart instance rather than an array of CartItems...

```suggestion
   * @param cartItems - The new cart items.
```

We could make it more wordy to clarify that this _**replaces any existing items in the cart**_ (at least I assume that's what it would do), but I think changing the param name/description should be enough.